common/src/service/ExceptionHandlingHelper.cs

```diff
@@ -120,7 +128,11 @@ public static async Task<ErrorCode> GetExceptionCodeAsync(HttpResponseMessage re
 
                 try
                 {
-                    Dictionary<string, string> messageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
+                    var messageFields = new Dictionary<string, string>();
+                    if (responseContent != null && responseContent.Message != null)
+                    {
+                        messageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
+                    }
 
                     if (messageFields != null
```

Comment:
That's what I thought before, but what I learnt by running this [test](https://github.com/Azure/azure-iot-sdk-csharp/blob/main/iothub/service/tests/ExceptionHandlingHelperTests.cs#L56) is that, in case of deserializing an empty `responseContent.Message`, the generated `messageFields` can be null, so we still need null check for this here.